include/dr/details/segments_tools.hpp

```diff
@@ -200,7 +207,8 @@ auto segments_(V &&v) {
 }
 
 template <rng::range V>
-  requires(lib::is_zip_view_v<std::remove_cvref_t<V>>)
+  requires(lib::is_zip_view_v<std::remove_cvref_t<V>> &&
+           distributed_zip_reference<rng::range_reference_t<V>>)
```

Comment:
@rscohn2 Is this the right condition for `segments` for `rng::zip_view`?

(Segments need to be disabled if it's not a zip of distributed ranges.)

Reply:
I deleted this code in the main branch today. The new code uses a member function, not an ADL.
https://github.com/oneapi-src/distributed-ranges/blob/db40f4c266ee993a0c8f6c606c3d8fd30463a4cc/include/dr/mhp/views/zip.hpp#L30